examples/apps/colorapp/src/colorteller/deploy.sh

```diff
@@ -14,12 +13,25 @@ if [ -z $AWS_DEFAULT_REGION ]; then
 fi
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
-COLOR_TELLER_IMAGE=${COLOR_TELLER_IMAGE:-"${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/colorteller"}
+ECR_URL="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com"
+COLOR_TELLER_IMAGE=${COLOR_TELLER_IMAGE:-"${ECR_URL}/colorteller"}
 GO_PROXY=${GO_PROXY:-"https://proxy.golang.org"}
+AWS_CLI_VERSION=$(aws --version 2>&1 | cut -d/ -f2 | cut -d. -f1)
+
+ecr_login() {
+        if [ $AWS_CLI_VERSION -eq 1 ]; then
+            $(aws ecr get-login --no-include-email)
+        elif [ $AWS_CLI_VERSION -eq 2 ]; then
+            aws ecr get-login-password | docker login --username AWS --password-stdin ${ECR_URL}
+        else
+            echo "Invalid AWS CLI version"
```

Comment:
Assuming my first comment is wrong: Can we call this "Unsupported AWS CLI version" instead? :)

Reply:
Agreed, I had just cut and pasted what was there but this is improved. Fixed them all up, thanks!